### PR TITLE
tracing: Provide default configuration when no host specified for k8s ingress

### PR DESCRIPTION
**Pull request by objectiser**
As requested in https://github.com/istio/istio/pull/9930#pullrequestreview-189617339, this PR supercedes #9930 for release-1.1 branch.

Additionally, due to the context being provider specific, have removed the Jaeger specific ingress and instead uses the `provider` value to be the default context. This works for both Jaeger and Zipkin, and follows the convention from #10699 which uses context paths related to the component (e.g. grafana and prometheus).

User can still define their own context path (e.g. `tracing.contextPath`), which would then be used for the Jaeger UI as well (instead of the previous `tracing.jaeger.contextPath` value), but Zipkin doesn't appear to be configurable in this area.

### CODE REVIEW

File: install/kubernetes/helm/subcharts/tracing/templates/deployment-jaeger.yaml [hunk 1]
```diff
@@ -56,10 +56,8 @@ spec:
             value: "9411"
           - name: MEMORY_MAX_TRACES
             value: "{{ .Values.jaeger.memory.max_traces }}"
-{{- if .Values.jaeger.contextPath }}
           - name: QUERY_BASE_PATH
-            value: {{ .Values.jaeger.contextPath }}
-{{- end }}
+            value: {{ if $.Values.contextPath }} {{ $.Values.contextPath }} {{ else }} /{{ $.Values.provider }} {{ end }}
```
Comment (morvencao) [inline install/kubernetes/helm/subcharts/tracing/templates/deployment-jaeger.yaml]: `$values` here can be simplified to `.values`

File: install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml [hunk 2]
```diff
@@ -17,16 +15,26 @@ metadata:
     {{- end }}
 spec:
   rules:
+{{- if .Values.ingress.hosts }}
     {{- range $host := .Values.ingress.hosts }}
     - host: {{ $host }}
       http:
         paths:
-          - path: /
+          - path: {{ if $.Values.contextPath }} {{ $.Values.contextPath }} {{ else }} /{{ $.Values.provider }} {{ end }}
```
Comment (morvencao) [inline install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml]: $values here can also be simplified to .values

File: install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml [hunk 3]
```diff
@@ -17,16 +15,26 @@ metadata:
     {{- end }}
 spec:
   rules:
+{{- if .Values.ingress.hosts }}
     {{- range $host := .Values.ingress.hosts }}
     - host: {{ $host }}
       http:
         paths:
-          - path: /
+          - path: {{ if .Values.contextPath }} {{ .Values.contextPath }} {{ else }} /{{ .Values.provider }} {{ end }}
```
Comment (morvencao) [inline install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml]: If tracing provider is zipkin, the path would be /zipkin, does zipkin support contextpath now??

Comment (objectiser) [reply install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml]: Yes https://github.com/openzipkin/zipkin/pull/1627. It worked for me - but would be good if someone else could also test it out to confirm.

Comment (sebastien-prudhomme) [reply install/kubernetes/helm/subcharts/tracing/templates/ingress.yaml]: @objectiser removing $ in a "range" loop breaks the Helm chart.

**Pull request merged**